STX.REST.RESTFulSense.Clients/Services/Foundations/HttpExchanges/HttpExchangeService.cs

```diff
@@ -0,0 +1,82 @@
+// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using STX.REST.RESTFulSense.Clients.Brokers.Https;
+using STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges;
+
+namespace STX.REST.RESTFulSense.Clients.Services.Foundations.HttpExchanges
+{
+    internal partial class HttpExchangeService : IHttpExchangeService
+    {
+        private readonly IHttpBroker httpBroker;
+
+        internal HttpExchangeService(IHttpBroker httpClientBroker)
+        {
+            this.httpBroker = httpClientBroker;
+        }
+
+        public ValueTask<HttpExchange>  GetAsync(
+            HttpExchange httpExchange,
+            CancellationToken cancellationToken = default) =>
+        TryCatch(async () =>
+        {
+            HttpMethod defaultHttpMethod = HttpMethod.Get;
+            Version defaultHttpVersion = HttpVersion.Version11;
+
+            HttpVersionPolicy defaultHttpVersionPolicy =
+                HttpVersionPolicy.RequestVersionOrLower;
+
+            ValidateHttpExchange(
+                httpExchange,
+                defaultHttpMethod,
+                defaultHttpVersion,
+                defaultHttpVersionPolicy);
```

Comment:
Please rename this to ValidateOnGet as the validation does more than validating the HttpExchange

```suggestion
            ValidateOnGet(
                httpExchange,
                defaultHttpMethod,
                defaultHttpVersion,
                defaultHttpVersionPolicy);
```

Reply:
Resolved

Reply:
It is not resolved until you push the changes.

Reply:
ok. gotcha. I was trying to keep track of them as I'm resolving them on the fly

Reply:
Resolved